server/src/main/java/org/opentosca/toscana/plugins/cloudformation/CloudFormationLifecycle.java

```diff
@@ -1,33 +1,66 @@
 package org.opentosca.toscana.plugins.cloudformation;
 
 import java.io.IOException;
+import java.util.Set;
 
+import org.opentosca.toscana.core.plugin.PluginFileAccess;
 import org.opentosca.toscana.core.transformation.TransformationContext;
+import org.opentosca.toscana.model.EffectiveModel;
+import org.opentosca.toscana.model.node.Compute;
+import org.opentosca.toscana.model.node.RootNode;
+import org.opentosca.toscana.model.visitor.VisitableNode;
+import org.opentosca.toscana.plugins.cloudformation.visitor.CloudFormationNodeVisitor;
 import org.opentosca.toscana.plugins.lifecycle.AbstractLifecycle;
 
 public class CloudFormationLifecycle extends AbstractLifecycle {
+    private final EffectiveModel model;
 
     public CloudFormationLifecycle(TransformationContext context) throws IOException {
         super(context);
+        model = context.getModel();
     }
 
     @Override
     public boolean checkModel() {
-        throw new UnsupportedOperationException();
+//        TODO implement model checks
+        return true;
     }
 
     @Override
     public void prepare() {
-        throw new UnsupportedOperationException();
+//        TODO implement preparation
     }
 
     @Override
     public void transform() {
-        throw new UnsupportedOperationException();
+        logger.info("Begin transformation to CloudFormation.");
+        PluginFileAccess fileAccess = context.getPluginFileAccess();
+        CloudFormationModule cfnModule = new CloudFormationModule(fileAccess);
+        Set<RootNode> nodes = model.getNodes();
+
+        try {
+            CloudFormationNodeVisitor cfnNodeVisitor = new CloudFormationNodeVisitor(logger, cfnModule);
+            for (VisitableNode node : nodes) {
+                if (node instanceof Compute) {
+                    node.accept(cfnNodeVisitor);
+                }
+            }
+            for (VisitableNode node : nodes) {
+                if (!(node instanceof Compute)) {
+                    node.accept(cfnNodeVisitor);
+                }
+            }
+
+            fileAccess.access("output/template.yaml").appendln(cfnModule.toString()).close();
+            logger.info("Transformation to CloudFormation successful.");
+        } catch (Exception e) {
+            logger.error("Transformation to CloudFormation unsuccessful. Please check the StackTrace for more Info.");
+            e.printStackTrace();
+        }
     }
 
     @Override
     public void cleanup() {
-        throw new UnsupportedOperationException();
+//        TODO implement cleanup
```

Comment:
Hust put NOOP there. I don't think the cleanup phase is specifically implemented by itself.

you might add something here if its needed.